Bottom: e38edb9a0acc6cfa5f5dbdc45d89aac2bf5729c0
Top:    d345ffc9c640135ca0de6638a27aeb8fcdaf72f9
Author: Jason Wang <jasowang@redhat.com>
Date:   2018-07-04 21:41:57 +0800

Refresh of exp

---

diff --git a/drivers/net/tun.c b/drivers/net/tun.c
index c67db25..d97d695 100644
--- a/drivers/net/tun.c
+++ b/drivers/net/tun.c
@@ -1579,6 +1579,7 @@ static void tun_rx_batched(struct tun_struct *tun, struct tun_file *tfile,
 	if (!more || tfile->rx_batched == rx_batched) {
 		INIT_LIST_HEAD(&list);
 		list_splice_tail_init(&tfile->rx_list, &list);
+		list_add_tail(&skb->list, &list);
 		rcv = true;
 		tfile->rx_batched = 0;
 	} else {
@@ -1590,8 +1591,6 @@ static void tun_rx_batched(struct tun_struct *tun, struct tun_file *tfile,
 	if (rcv) {
 		struct sk_buff *nskb, *tmp;
 
-		list_add_tail(&skb->list, &list);
-
 		local_bh_disable();
 		list_for_each_entry_safe(nskb, tmp, &list, list) {
 			list_del_init(&nskb->list);
